xbmc/pvr/PVRDatabase.cpp

```diff
@@ -661,8 +644,23 @@ bool CPVRDatabase::UpdateLastWatched(const CPVRChannel &channel)
 
 bool CPVRDatabase::UpdateLastWatched(const CPVRChannelGroup &group)
 {
-  std::string strQuery = PrepareSQL("UPDATE channelgroups SET iLastWatched = %d WHERE idGroup = %d",
+  const std::string strQuery = PrepareSQL("UPDATE channelgroups SET iLastWatched = %d WHERE idGroup = %d",
     group.LastWatched(), group.GroupID());
 
   return ExecuteQuery(strQuery);
 }
+
+int CPVRDatabase::GetChannelId(int iClientId, int iUniqueId)
+{
+  int channelId = -1;
+
+  const std::string strQuery = PrepareSQL("SELECT idChannel FROM channels WHERE iClientId = %d AND iUniqueId = %d",
+                                          iClientId, iUniqueId);
+
+  std::string value = GetSingleValue(strQuery);
```

Comment:
There is no need to convert the int of the db to a string and back to an int.
Please change the code to:

```
if (ResultQuery(strQuery) && !m_pDS->eof())
  channelId = m_pDS->fv("idChannel").get_asInt()
```